packages/adminrouter/extra/src/test-harness/modules/generic_test_code/common.py

```diff
@@ -97,6 +98,67 @@ def generic_response_headers_verify_test(
             header_is_absent(resp.headers.items(), name)
 
 
+def generic_upstream_cookies_verify_test(
+        ar,
+        headers,
+        path,
+        cookies_to_send=None,
+        assert_cookies_present=None,
+        assert_cookies_absent=None):
+    """Test if cookies that are passed to the upstream by AR are correct
+
+    Helper function meant to simplify writing multiple tests testing the
+    same thing for different endpoints.
+
+    Arguments:
+        ar: Admin Router object, an instance of runner.(ee|open).Nginx
+        headers (dict): headers dict that contains JWT and cookies. The auth
+            data it contains must be valid.
+        path (str): path for which request should be made
+        cookies_to_send (dic): dictionary containing all the cookies that should
+            be send in the request
+        assert_cookies_present (dict): cookies to test where key is the
+            asserted cookie name and value is expected value of the cookie
+        assert_cookies_absent (dict): cookies that *MUST NOT* be present in the
```

Comment:
This actually expects a `list()` or `set()` at the moment, which is great, the docstring just needs to get updated.